firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -162,51 +162,38 @@ public void onFailure(@NonNull @NotNull Exception e) {
     checkForUpdateTaskCompletionSource =
         new TaskCompletionSource<>(checkForUpdateCancellationSource.getToken());
 
-    firebaseInstallationsApi
-        .getId()
+    Task<String> installationIdTask = firebaseInstallationsApi.getId();
+    Task<InstallationTokenResult> installationAuthTokenTask =
+        firebaseInstallationsApi.getToken(true);
+
+    Tasks.whenAllComplete(installationIdTask, installationAuthTokenTask)
         .addOnSuccessListener(
-            new OnSuccessListener<String>() {
-              @Override
-              public void onSuccess(String fid) {
-                firebaseInstallationsApi
-                    .getToken(true)
-                    .addOnSuccessListener(
-                        new OnSuccessListener<InstallationTokenResult>() {
+            tasks -> {
+              String fid = (String) tasks.get(0).getResult();
```

Comment:
Out of curiosity, does installationIdTask.getResult() throw an error?

Reply:
For some reason I thought it would but it doesn't, changing it to that.